public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Seg.tsx

```diff
@@ -90,41 +96,70 @@ const SelReload = ({ loadOptions, allowCustomValue, onChange, onClose }: SelRelo
 };
 
 type SelSingleLoadProps = {
+  value: string;
   loadOptions: (filter: string) => Promise<SelVal[]>;
   onClose: () => void;
-  onChange: (v: SelVal) => void;
+  onChange: (v: SelVal | null) => void;
   allowCustomValue?: boolean;
+  isClearable: boolean;
 };
 
-const SelSingleLoad = ({ loadOptions, allowCustomValue, onChange, onClose }: SelSingleLoadProps): JSX.Element => {
+const SelSingleLoad = ({
+  value,
+  loadOptions,
+  allowCustomValue,
+  isClearable,
+  onChange,
+  onClose,
+}: SelSingleLoadProps): JSX.Element => {
   const [loadState, doLoad] = useAsyncFn(loadOptions, [loadOptions]);
 
   useEffect(() => {
     doLoad();
   }, [doLoad, loadOptions]);
 
+  const options = loadState.value ?? [];
+
+  // if the current-value is not in the list-of-options, and it is not empty, we have to add
+  // it to the options
+  const valueInOptions = options.some((x) => x.value === value);
+
+  const fullOptions = value === '' || valueInOptions ? options : [{ label: value, value }, ...options];
```

Comment:
what if for the "empty string" option we show something different? from the UI it's not really clear I'm selecting an empty string.

maybe something like:

![Screenshot 2021-05-19 at 12 26 10](https://user-images.githubusercontent.com/1170767/118805441-d18d3280-b89d-11eb-895b-468c6219a600.png)

although I don't know if it might generate confusion if you actually have a `""` in your metrics, i'd loop in someone from UX